Blockbase: Fix color customizations

scruffian opened the pull request:
<!-- Thanks for contributing to our free themes! Please provide as much information as possible with your Pull Request by filling out the following - this helps make reviewing much quicker! -->

#### Changes proposed in this Pull Request:
When you change other cusomizer settings, the color settings are lost. This adds some code to preserve the color settings when others are changed.

For testing instructions see #4112

#### Related issue(s):
Fixes #4112

scruffian committed:
Blockbase: Fix color customizations

pbking reviewed (approved):
👍 This checks out. I couldn't break it.
🚢

scruffian committed:
remove unneeded changes

scruffian commented:
Update: I updated this to also work with Font customizations. I found a bug which I also fixed. To reproduce...

- Set the font to be "Bodoni Mona"
- Save
- Reload the page
- Change the font to Arvo - the preview updates
- Update the Site Title
- Change the font back to "Bodoni Mona"
- Nothing happens....

This is fixed in this PR...

scruffian committed:
remove unused variable

pbking reviewed (approved):
Bug reproduced in /trunk and confirmed fixed here.

Color changes stick around and persist as expected.

No PHP errors.

Checks out again.

I re-approve!
